docs/concepts/validators.md

```diff
@@ -1,635 +1,575 @@
+In its simplest form, a field validator is a callable taking the value to be validated as an argument and
+returning the validated value. The callable can perform checks for specific conditions (see
+[raising validation errors](#raising-validation-errors)) and make changes to the validated value.
 
-```python
-from typing import Any, List
+**Four** different types of validators can be used. They can all be defined using the
+[annotated pattern](./fields.md#the-annotated-pattern) or using the
+[`field_validator()`][pydantic.field_validator] decorator, applied on a [class method][classmethod]:
 
-from typing_extensions import Annotated
+[](){#field-after-validator}
 
-from pydantic import BaseModel, ValidationError
-from pydantic.functional_validators import AfterValidator
+- __*After* validators__: they run after Pydantic's internal validation. They are generally more type safe and thus easier to implement.
 
+    === "Annotated pattern"
 
-def check_squares(v: int) -> int:
-    assert v**0.5 % 1 == 0, f'{v} is not a square number'
-    return v
+        Here is an example of a validator performing a validation check, and returning the value unchanged.
 
+        ```python
+        from typing_extensions import Annotated
 
-def double(v: Any) -> Any:
-    return v * 2
+        from pydantic import AfterValidator, BaseModel, ValidationError
 
 
-MyNumber = Annotated[int, AfterValidator(double), AfterValidator(check_squares)]
+        def is_even(value: int) -> int:
+            if value % 2 == 1:
+                raise ValueError(f'{value} is not an even number')
+            return value  # (1)!
 
 
-class DemoModel(BaseModel):
-    number: List[MyNumber]
+        class Model(BaseModel):
+            number: Annotated[int, AfterValidator(is_even)]
 
 
-print(DemoModel(number=[2, 8]))
-#> number=[4, 16]
-try:
-    DemoModel(number=[2, 4])
-except ValidationError as e:
-    print(e)
-    """
-    1 validation error for DemoModel
-    number.1
-      Assertion failed, 8 is not a square number
-    assert ((8 ** 0.5) % 1) == 0 [type=assertion_error, input_value=4, input_type=int]
-    """
-```
+        try:
+            Model(number=1)
+        except ValidationError as err:
+            print(err)
+            """
+            1 validation error for Model
+            number
+              Value error, 1 is not an even number [type=value_error, input_value=1, input_type=int]
+            """
+        ```
 
-In this example we used some type aliases (`MyNumber = Annotated[...]`).
-While this can help with legibility of the code, it is not required, you can use `Annotated` directly in a model field type hint.
-These type aliases are also not actual types but you can use a similar approach with `TypeAliasType` to create actual types.
-See [Custom Types](../concepts/types.md#custom-types) for a more detailed explanation of custom types.
+        1. Note that it is important to return the validated value.
```

Comment:
I think this is worth an admonition, generally, in the validators section. Keeping the annotation is good as well. I think a little bit of redundancy here doesn't hurt.

Reply:
We already state

> In its simplest form, a field validator is a callable taking the value to be validated as an argument and
returning the validated value.

I can maybe emphasize on _returning the validated value_.

Reply:
Let's emphasize, and if we get more questions in issues, let's add an admonition.